src/simplify_algebra.cpp

```diff
@@ -440,11 +440,44 @@ struct find_mul_add
         auto x_ins = r.instructions["x"];
         assert(x_ins != b_ins);
 
+        if(a_ins->get_shape().scalar())
```

Comment:
This should also check that it goes into convolution or gemm for the exception. It should also be done in the matcher.